research/conditional/utils/conditional_trainer.py

```diff
@@ -210,22 +210,16 @@ def _decode_samples(self, step):
                 iteration=step,
             )
 
-    def _optimize(self, loss, should_apply_gradient=False):
-        # since we sum gradients averaged over multiple smaller batches, we need to normalize here
-        loss /= self.gradient_accumulation_steps
-        if self.gradient_accumulation_steps == 1:
-            self.optimizer.zero_grad()
-        # clear computation graph, store gradients
-        self.scaler.scale(loss).backward()
```

Comment:
what happened here? how is gradient accumulation implemented now?

Reply:
It zeros out gradients after applying them, about 10 lines below. If previous logic was required (zeroing before if no accumulation and after if accumulation) i can try to revert it back

Reply:
As discussed in person, this is acceptable, even better than before.